### PR TITLE
Feature/secretary controller

### PR DESCRIPTION
**Pull request by x1mx1x**

### CODE REVIEW

File: CharlieBackend.Api/Controllers/SecretariesController.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            var createdSecretaryDto = await _secretaryService.CreateSecretaryAsync(secretaryDto);
+
+            if (createdSecretaryDto == null)
+            {
+                return StatusCode(422, "Cannot create secretary.");
+            }
+
+            return Ok(new { createdSecretaryDto.Id });
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/SecretariesController.cs:44]: Return a created model.

File: CharlieBackend.Api/Controllers/SecretariesController.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        {
+
+            var secretariesDtos = await _secretaryService.GetAllSecretariesAsync();
+
+            return Ok(secretariesDtos);
+        }
+
+        [Authorize(Roles = "4")]
+        [HttpPut("{id}")]
+        public async Task<ActionResult> updateSecretary(long id, UpdateSecretaryDto secretaryDto)
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/SecretariesController.cs:59]: Rename to UpdateSecretary.

File: CharlieBackend.Business/Services/SecretaryService.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                if (!string.IsNullOrEmpty(secretaryDto.Password))
+                {
+                    foundSecretary.Account.Salt = _accountService.GenerateSalt();
+                    foundSecretary.Account.Password = _accountService.HashPassword(secretaryDto.Password, foundSecretary.Account.Salt);
+                }
+
+                await _unitOfWork.CommitAsync();
+
+                return _mapper.Map<SecretaryDto>(foundSecretary);
+
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/SecretaryService.cs:109]: Remove this new line.

File: CharlieBackend.Data/Repositories/Impl/SecretaryRepository.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                            => secretary.AccountId == accountId);
+        }
+
+        public new Task<Secretary> GetByIdAsync(long id)
+        {
+            return _applicationContext.Secretaries
+                .Include(secretary => secretary.Account)
+                .FirstOrDefaultAsync(secretary => secretary.Id == id);
+        }
+
```
Comment (SerjKocherga) [inline CharlieBackend.Data/Repositories/Impl/SecretaryRepository.cs:35]: Remove this empty lines.

File: CharlieBackend.Data/Repositories/Impl/SecretaryRepository.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    }
+}
+
```
Comment (SerjKocherga) [inline CharlieBackend.Data/Repositories/Impl/SecretaryRepository.cs:39]: 39 and 40 lines also should be removed.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]

**Review by SnipeFrost — APPROVED** [review]
good

**Review by SerjKocherga — APPROVED** [review]

File: CharlieBackend.Business/Services/SecretaryService.cs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                        credsSent = false;
+                        return _mapper.Map<SecretaryDto>(secretary);
+                        //need to handle the exception with a right logic to sent it for contorller if fails
+                        //throw new System.Exception("Faild to send credentials");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    transaction.Rollback();
+                    Console.WriteLine("Debug info exception details: " + ex);
```
Comment (sober607) [inline CharlieBackend.Business/Services/SecretaryService.cs:78]: This is for debug stage only

**Review by sober607 — COMMENTED** [review]
Small change recommended

**Pull request merged**